extensions/panache/hibernate-orm-panache/deployment/src/main/java/io/quarkus/hibernate/orm/panache/deployment/PanacheJpaRepositoryEnhancer.java

```diff
@@ -51,24 +51,26 @@ protected String getPanacheOperationsBinaryName() {
 
         @Override
         public void visitEnd() {
-            // Bridge for findById
-            MethodVisitor mv = super.visitMethod(Opcodes.ACC_PUBLIC | Opcodes.ACC_SYNTHETIC | Opcodes.ACC_BRIDGE,
-                    "findById",
-                    "(Ljava/lang/Object;)Ljava/lang/Object;",
-                    null,
-                    null);
-            mv.visitParameter("id", 0);
-            mv.visitCode();
-            mv.visitIntInsn(Opcodes.ALOAD, 0);
-            mv.visitIntInsn(Opcodes.ALOAD, 1);
-            mv.visitMethodInsn(Opcodes.INVOKEVIRTUAL,
-                    daoBinaryName,
-                    "findById",
-                    "(Ljava/lang/Object;)" + entitySignature, false);
-            mv.visitInsn(Opcodes.ARETURN);
-            mv.visitMaxs(0, 0);
-            mv.visitEnd();
-
+            // Bridge for findById, but only if we actually know the end entity (which we don't for intermediate
+            // abstract repositories that haven't fixed their entity type yet
+            if (!"Ljava/lang/Object;".equals(entitySignature)) {
```

Comment:
Why don't you just avoid any enahcement for abstract repositories/entities ?
This way it will speed up the build process no ?
And you will also be able to remove some cheks inside the processor as PanacheEntity itself is abstract

Reply:
Because this also happens for non-abstract classes, as my test shows.

Reply:
OK